tree/ntuple/v7/src/RPageStorageS3.cxx

```diff
@@ -0,0 +1,422 @@
+/// \file RPageStorageS3.cxx
+/// \ingroup NTuple ROOT7
+/// \author Max Orok <maxwellorok@gmail.com>
+/// \date 2021-06-03
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2021, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RCluster.hxx>
+#include <ROOT/RClusterPool.hxx>
+#include <ROOT/RColumn.hxx>
+#include <ROOT/RPageAllocator.hxx>
+#include <ROOT/RPagePool.hxx>
+#include <ROOT/RPageStorageS3.hxx>
+
+#include <iostream>
+#include <string>
+
+#include "davix.hpp"
+
+namespace {
+   static const std::string kOidAnchor = std::to_string(std::uint64_t(-1));
+   static const std::string kOidHeader = std::to_string(std::uint64_t(-2));
+   static const std::string kOidFooter = std::to_string(std::uint64_t(-3));
+} // anonymous namespace
+
+namespace ROOT {
+namespace Experimental {
+namespace Detail {
+
+class RS3Handle {
```

Comment:
Maybe change the name of the class, e.g. `RS3Bucket`, given that it provides read/write access to object in a bucket.